src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -227,35 +243,46 @@ ParsedQuery::GraphPattern Visitor::visitTypesafe(
   }
 }
 
-// ____________________________________________________________________________________
-namespace {
-template <typename A, typename B>
-std::pair<vector<A>, vector<B>> splitUp(vector<std::variant<A, B>> elems) {
-  vector<A> as;
-  vector<B> bs;
-  auto aOp = [&as](A a) { as.push_back(std::move(a)); };
-  auto bOp = [&bs](B b) { bs.push_back(std::move(b)); };
-  for (auto& elem : elems) {
-    std::visit(ad_utility::OverloadCallOperator{aOp, bOp}, std::move(elem));
+Visitor::OperationsAndFilters Visitor::visitTypesafe(
+    Parser::GroupGraphPatternSubContext* ctx) {
+  vector<GraphPatternOperation> ops;
+  vector<SparqlFilter> filters;
+
+  auto filter = [&filters](SparqlFilter filter) {
+    filters.emplace_back(std::move(filter));
+  };
+  auto op = [&ops](GraphPatternOperation op) {
+    ops.emplace_back(std::move(op));
+  };
+
+  if (ctx->triplesBlock()) {
+    ops.emplace_back(visitTypesafe(ctx->triplesBlock()));
+  }
+  auto others =
+      visitVector<std::pair<variant<GraphPatternOperation, SparqlFilter>,
+                            std::optional<BasicGraphPattern>>>(
+          ctx->graphPatternNotTriplesAndMaybeTriples());
```

Comment:
You have an alias for that very long type.
and add a `// TODO make `visitVector` deduce its return type automatically`

Reply:
Good reminder. I should have a (mostly) finished `visitVectorTypesafe` lying around.